### ISSUE TITLE
Including undertow extension disables devmode NotFound Exception Mapper

### ISSUE THREAD

**Issue by yspotts** [quarkusio/quarkus#25426]
### Describe the bug

2.8.2.Final

Dev mode includes a handy NotFound exception mapper that shows all the endpoints available
If one includes the `undertow` extension, this seems to disable/prevent this special error page

### Expected behavior

Should show the exception mapper (as it shows without the `undertow` extension)

![image](https://user-images.githubusercontent.com/887008/167177515-4fa7f9e4-89cc-4541-9044-b8496525a908.png)

### Actual behavior

Displays `Not Found` text

![image](https://user-images.githubusercontent.com/887008/167177765-f7b1f240-a8f8-473c-ae5f-978df4c519f4.png)

### How to Reproduce?

Steps to reproduce:

1. Create a new project with `quarkus-resteasy-reactive` and `quarkus-resteasy-reactive-jackson` extensions with sample code
2. Add the `undertow` extension
3. Navigate to a url that should produce a 404

### Output of `uname -a` or `ver`

Darwin Yoels-MacBook-Pro 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.14" 2022-01-18 OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05) OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05, mixed mode, sharing)

### GraalVM version (if different from Java)

22.0.0.2.r11

### Quarkus version or git rev

2.8.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

7.4.1

**Comment by geoand** [comment]
This is kind of a weird use case. The reason it's done this way is because when RESTEasy Reactive and Undertow are both present, the latter is meant to be a "fallback" of sorts.

I suppose we could add a property that would override this behavior...

**Comment by yspotts** [comment]
Agreed its an edge case. My specific use case is that I want to enable the h2 UI console and my understanding is that the only way to do so is to enable `undertow` and `vertx` extensions and enable the h2 console servlet in `web.xml`.

**Comment by geoand** [comment]
Hm... If that's the case, then we should probably come up with a better way to support it.

@gsmet do you know what the H2 UI requires?

**Comment by geoand** [comment]
> This is kind of a weird use case. The reason it's done this way is because when RESTEasy Reactive and Undertow are both present, the latter is meant to be a "fallback" of sorts.
> 
> I suppose we could add a property that would override this behavior...

Closing as we have `quarkus.rest.resume-on-404`
